proxy/coco/src/source.rs

```diff
@@ -512,9 +512,35 @@ pub struct LocalState {
 ///
 /// Will return [`Error`] if the repository doesn't exist.
 pub fn local_state(repo_path: &str) -> Result<LocalState, Error> {
+    let repo = git2::Repository::open(repo_path)?;
+    let first_branch = repo
+        .branches(Some(git2::BranchType::Local))?
+        .filter_map(|branch_result| {
+            let branch = match branch_result {
+                Ok(s) => s,
```

Comment:
This syntax looks a little funny but i'm also a rust noob so 🤷‍♀️

Reply:
The `?` operator actually works on `Option` (magically coercing `NoneError` back to `None` where applicable):

```rust
let (branch, _) = branch_result.ok()?;
let name = branch.name().ok()?;
name.map(String::from)
```